lib/cli/objectlistcommand.cpp

```diff
@@ -104,6 +118,16 @@ int ObjectListCommand::Run(const boost::program_options::variables_map& vm, cons
 	Log(LogNotice, "cli")
 		<< "Parsed " << objects_count << " objects.";
 
+	auto objectsPathCtime (GetCtime(Configuration::ObjectsPath));
+	auto varsPathCtime (GetCtime(Configuration::VarsPath));
```

Comment:
Reading these at the end might lead to incorrect warnings if there's a concurrent config deployment for example. Would it be possible to fetch the timestamps using `fstat` when reading the files?

Reply:
Which "concurrent config deployment"? #9267

Reply:
Config deployment (be it `icinga2 daemon -C` run by the config package API or the daemon actually reloading) running concurrently with your call to `icinga2 object list`.

Reply:
> Reading these at the end might lead to incorrect warnings

Is this an actual problem?

> it won't be confusing if it says outdated by a few seconds.

Reply:
The common case is probably fine. If the daemon reloads while the `object list` command is running, the warning isn't even wrong.

However, if you run the daemon using `icinga2 daemon --dump-objects`, `icinga2 object list` might read an old file, the daemon could then update it and `object list` then uses the metadata of the updated file for the warning.

That could also be addressed by only allowing `--dump-objects` in combination with `-C`/`--validate`. (This would then still have a potential race condition with multiple people running `object list` and `daemon -C --dump-objects` at the same time.)

Reply:
> Would it be possible to fetch the timestamps using `fstat` when reading the files?

It's read via an fstream and IMAO you mention very special cases.

Reply:
Well, it would forbid letting the daemon auto-update the index on reload.

Reply:
That's the purpose. Because doing so has race-conditions with how the check is done currently.

And in general, I think it's a good idea to make updating `icinga2.cache` an explicit operation. If this is done randomly by the daemon (well, not exactly randomly, but unpredictably enough, i.e. you don't necessarily know when someone clicked  deploy in Director the last time), you don't really know the exact state that was dumped to the file. If you have to `daemon -C --dump-objects` explicitly, you know when this was done and see the exact state at that time. Also, this then takes objects into account that were changes using the API in the meantime, which is a good thing (which is why you were able to remove that specific part in the documentation).

Reply:
And why you think that given the considerations from Julian?

Reply:
The "problems" Julian described aren't there by default. One has to enable `daemon --dump-objects` explicitly if desired and to know what they're doing.

Reply:
> know what they're doing

In this context, this would mean that they should know that they are enabling a code path that sometimes may skip a warning.

Reply:
I'm still not convinced. You mention very special cases which aren’t actual problems.

Reply:
Running the actual daemon with `--dump-objects` is something I hope nobody will ever do: you get a performance penalty with every reload and if you want to know things about the running daemon, you can use the API. Not allowing it in the first place ensures we don't end up in some "well, there's this strange feature, we hope nobody does this but we can't tell for sure" situation.

Reply:
No. I don’t hope this. But I can restrict it if you really want to.